geopandas/io/sql.py

```diff
@@ -96,3 +97,291 @@ def load_geom_text(x):
+            sql = "SELECT UpdateGeometrySRID('{schema}','{tbl}','{geom}',{crs})".format(
+                schema=tbl.table.schema, tbl=tbl.table.name, geom=geom_name, crs=0
+            )
+            cur.execute(sql)
+
+        if chunksize is None:
+            _populate_db(gdf, conn, cur, index, tbl)
+        else:
+            # Insert in chunks
+            chunks = _get_chunks(gdf, chunksize)
+
+            for chunk in chunks:
+                _populate_db(chunk, conn, cur, index, tbl)
+
+        # SRID needs to be updated afterwards as Shapely does not support
+        # EWKT/EWKB geometries, see:
+        # https://community.gispython.narkive.com/qTVQCl3f/ewkt-ewkb-support-in-shapely
+        sql = "SELECT UpdateGeometrySRID('{schema}','{tbl}','{geom}',{srid})".format(
+            schema=tbl.table.schema, tbl=tbl.table.name, geom=geom_name, srid=srid
+        )
+        cur.execute(sql)
+        conn.commit()
+
+    except Exception as e:
+        conn.connection.rollback()
+        raise e
+    finally:
+        conn.close()
+
+
+def write_postgis(
+    gdf,
+    name,
+    con,
+    schema=None,
+    if_exists="fail",
+    index=False,
+    index_label=None,
+    chunksize=None,
+    dtype=None,
+):
+    """
+    Upload GeoDataFrame into PostGIS database.
+
+    Parameters
+    ----------
+    name : str
+        Name of the target table.
+    con : sqlalchemy.engine.Engine
+        Active connection to the PostGIS database.
+    if_exists : {‘fail’, ‘replace’, ‘append’}, default ‘fail’
+        How to behave if the table already exists.
+          - fail: Raise a ValueError.
+          - replace: Drop the table before inserting new values.
+          - append: Insert new values to the existing table.
+    schema : string, optional
+        Specify the schema. If None, use default schema: 'public'.
+    index : bool, default True
+        Write DataFrame index as a column.
+        Uses *index_label* as the column name in the table.
+    index_label : string or sequence, default None
+        Column label for index column(s).
+        If None is given (default) and index is True,
+        then the index names are used.
+    chunksize : int, optional
+        Rows will be written in batches of this size at a time.
+        By default, all rows will be written at once.
+    dtype : dict of column name to SQL type, default None
+        Specifying the datatype for columns.
+        The keys should be the column names and the values
+        should be the SQLAlchemy types.
+    """
+    from geoalchemy2 import Geometry
+
+    gdf = gdf.copy()
+    geom_name = gdf.geometry.name
+    if schema is not None:
+        schema_name = schema
+    else:
+        schema_name = "public"
+
+    # Get srid
+    srid = _get_srid_from_crs(gdf)
+
+    # Get geometry type and info whether data contains LinearRing
+    geometry_type, has_curve = _get_geometry_type(gdf)
+
+    # Build dtype with Geometry (srid is updated afterwards)
```

Comment:
Why can't we specify the srid here when creating the type with geoalchemy?

Reply:
Hmm, I'm not 100 % sure what you mean here? If you mean that why don't we use the `srid` parameter of [_GISType](https://geoalchemy-2.readthedocs.io/en/latest/types.html#geoalchemy2.types._GISType) object, it is because Shapely does not support EWKT or EWKB formats: https://community.gispython.narkive.com/qTVQCl3f/ewkt-ewkb-support-in-shapely